cylc/flow/network/graphql.py

```diff
@@ -128,200 +149,88 @@ def strip_null(data):
-    # Search request document to determine if 'stripNull: true' is set
-    # as and argument. It can not be done in the middleware, as they
-    # can be Promises/futures (so may not been resolved at this point).
-    variable_values = kwargs['variable_values'] or {}
-    doc_args = AstDocArguments(schema, document_ast, variable_values)
-    if doc_args.has_arg_val(STRIP_ARG, True):
-        if kwargs.get('return_promise', False) and hasattr(result, 'then'):
-            return result.then(null_stripper)  # type: ignore[union-attr]
-        return null_stripper(result)
-    return result
-
-
-class CylcGraphQLBackend(GraphQLBackend):
-    """Return a GraphQL document using the default
-    graphql executor with optional null-stripping of result.
-
-    The null value stripping of result is triggered by the presence
-    of argument & value "stripNull: true" in any field.
+    if is_awaitable(exe_result):
+        return async_next(strip_null, exe_result)
+    return strip_null(exe_result)
 
-    This is a modification of GraphQLCoreBackend found within:
-        https://github.com/graphql-python/graphql-core-legacy
-    (graphql-core==2.3.2)
 
-    Args:
-
-        executor (object): Executor used in evaluating the resolvers.
+class CylcVisitor(Visitor):
+    """Traverse graphql document/query to find an argument in a given state.
 
+    Find whether an argument is set to a specific value anywhere in the
+    document (i.e. 'strip_null' set to 'True'), and stop on the first
+    occurrence.
     """
+    def __init__(self, type_info, variable_values, doc_arg) -> None:
+        super().__init__()
+        self.type_info = type_info
+        self.variable_values = variable_values
+        self.doc_arg = doc_arg
+        self.arg_flag = False
+
+    def enter(self, node, key, parent, path, ancestors):
+        if hasattr(node, 'arguments'):
+            field_def = self.type_info.get_field_def()
+            arg_vals = get_argument_values(
+                field_def,
+                node,
+                self.variable_values
+            )
+            if arg_vals.get(self.doc_arg['arg']) == self.doc_arg['val']:
+                self.arg_flag = True
+                return self.BREAK
+        return self.IDLE
 
-    def __init__(self, executor=None):
-        self.execute_params = {"executor": executor}
-
-    def document_from_string(self, schema, document_string):
-        """Parse string and setup request document for execution.
-
-        Args:
+    def leave(self, node, key, parent, path, ancestors):
+        return self.IDLE
 
-            schema (graphql.GraphQLSchema):
-                Schema definition object
-            document_string (str):
-                Request query/mutation/subscription document.
 
-        Returns:
+class CylcExecutionContext(ExecutionContext):
 
-            graphql.GraphQLDocument
+    def execute_operation(
+        self, initial_result_record, root_value
```

Comment:
Pylance is flagging this as a potential problem

> Method "execute_operation" overrides class "ExecutionContext" in an incompatible manner
>
> Parameter 2 name mismatch: base parameter is named "operation", override parameter is named "initial_result_record"

Reply:
Well, I'm intentionally overwriting it... Could take a look I suppose

Reply:
Changed back to `operation` ..